### DISCUSSION TITLE
ProxyRack

### DISCUSSION THREAD

**Discussion by af0rte** [engageub/InternetIncome#112] [General]
i´m tryng to use ProxyRack but i have this error:

Update done!
Error: ENOENT: no such file or directory, unlink '/Roaming/PoP/log.log'
    at Object.unlinkSync (fs.js:1136:3)
    at _0x2c25ab (/app/script.js:1:1676538)
    at _0x384958 (/app/script.js:1:1676778)
    at Object.<anonymous> (/app/script.js:1:1680542)
    at Object.<anonymous> (/app/script.js:1:1680605)
    at _0x34ba2d (/app/script.js:1:181281)
    at /app/script.js:1:182218
    at Object.<anonymous> (/app/script.js:1:182251)
    at Module._compile (internal/modules/cjs/loader.js:999:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1027:10)
    at Module.load (internal/modules/cjs/loader.js:863:32)
    at Function.Module._load (internal/modules/cjs/loader.js:708:14)
    at Function.executeUserEntryPoint [as runMain] (internal/modules/run_main.js:60:12)
    at internal/main/run_main_module.js:17:47 {
  errno: -2,
  syscall: 'unlink',
  code: 'ENOENT',
  path: '/Roaming/PoP/log.log'
}
sending
[Error: ENOENT: no such file or directory, open './go_version.txt'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: './go_version.txt'
}
Error:  socket hang up
{ localVersion: '', remoteVersion: '56' }
downloading app
sending
Error:  socket hang up
{ success: true }
sending
writing 56 -> ./go_version.txt
written
go app started at: 1713220206

connecting to home: 1713220206

Error:  socket hang up
connected to home: 1713220206

sending
Error:  socket hang up
sending
Error:  socket hang up
sending
Error:  socket hang up
sending
Error:  socket hang up
sending
Error:  socket hang up
sending
Error:  socket hang up
sending
Error:  socket hang up
sending
Error:  socket hang up
sending
Error:  socket hang up
sending
Error:  socket hang up
sending
Error:  socket hang up
sending
Error:  socket hang up
sending
Error:  socket hang up
sending
Error:  socket hang up
sending
Error:  socket hang up

Help me!

Thanks!

**Comment by engageub** [comment]
Hello,
The Errors are generated by the container which are common. You may have to check with proxyrack owner.
It shows "connected to home: 1713220206" .
Please wait for 10 minutes and add it to your dashboard. If the device is not added, you can update here.

Thank you

**Reply by af0rte** [reply]
I can see now on dashboard i´ts online now :)

![image](https://github.com/engageub/InternetIncome/assets/36008040/91975df3-38a9-42a6-88a6-8dfa27e224a2)

Thanks for you help :)

**Comment by TradeBW** [comment]
Is there way to bulk delete Proxyrack devices via api ?

**Comment by engageub** [comment]
Proxyrack already has API to delete the device Id.
Please follow the api doc below to delete the devices.

https://peer.proxyrack.com/api-doc

Thank you

**Reply by TradeBW** [reply]
I am aware about this doc but  it doesn't have end point to fetch all the device ids in bulk.

**Comment by engageub** [comment]
Hi,

You may need to either contact Proxyrack Support or develop a JavaScript script to remove the IDs from the dashboard.
If performing this action through the browser, you can implement a loop to automatically click the delete button for each device listed on the dashboard.

Thank you

**Comment by TradeBW** [comment]
Why proxyrack.sh is removed from test branch ?

**Reply by engageub** [reply]
Proxyrack override has been removed due the the latest update from proxyrack team. They have modified the code to support these overrides. So this file is no longer required as the changes are present in their official docker image.

**Comment by TradeBW** [comment]
Because of this the test branch is failing, I dont think main branch supports docker deployment delay seconds.

**Comment by engageub** [comment]
Hi,
Could you please update now. 

Thank you

**Reply by TradeBW** [reply]
Test Branch working now, Thanks for the quick heads up. 
Docker delay will be supported in main branch by when ?

**Reply by engageub** [reply]
Hi,

The default delay in main branch after starting the proxy container is 1 second.
I was planning to merge test branch with main branch as it supports multi IP and multi VPN.
There were no issues reported with respect to  multi IP and multi VPN on test branch.
To add delay in main branch it requires changes in multiple places (38 lines of code) unlike test branch.
If you would like to use this feature, please continue using test branch for now.

Thank you

**Reply by TradeBW** [reply]
Aah, Test branch looks much fine then for now.
